app/assets/javascripts/SourceCodeGlower/SourceCodeLine.js

```diff
@@ -22,26 +25,203 @@ function SourceCodeLine(line_node) {
-SourceCodeLine.prototype.glow = function() {
-  // Increase the depth
+// Increase a Source Code Line's glow depth, add listeners
+// Splits nodes at start and end offsets if needed
+SourceCodeLine.prototype.glow = function(annotationId, start, end,
+  hoverOnFunction, hoverOffFunction) {
+  // Increase the depth and turn on design mode for highlighting
   this.incGlowDepth(1);
 
-  // Add the appropriate glow class
-  this.getLineNode().addClass('source_code_glowing_' + this.getGlowDepth());
+  // Save line node and get all text nodes in the line
+  var node = this.getLineNode();
+  var textNodes = this.getAllTextNodes(node);
+  if (end == -1){
+    end = this.getLineNode().textContent.length;
+  }
+
+  // Loop prep, for tracking nodes and character counts
+  var startNode = null;
+  var startNodeOffset = 0;
+  var endNode = null;
+  var endNodeOffset = 0;
+  var foundStart = false;
+  var currentCharCount = 0;
+
+  // Loop over nodes, add length of passed nodes, save start and end then break
+  for (var i = 0; i < textNodes.length; i++) {
+    // End char offset of the current text node
+    var endCharCount = currentCharCount + textNodes[i].length;
+
+    // If start is between current count and end of the current node, save it
+    if (!foundStart && start >= currentCharCount && start <= endCharCount) {
+      foundStart = true;
+
+      // Will need to split the node if the glow starts inside of it
+      if (start != endCharCount) {
+        // Save reference to span node, create span wrapper if blank space
+        if (textNodes[i].parentNode.parentNode === node) {
+          startNode = document.createElement("span");
+          startNode.innerHTML = textNodes[i].textContent;
+          textNodes[i].parentNode.replaceChild(startNode, textNodes[i]);
+          textNodes[i] = startNode.childNodes[0];
+        }
+        else {
+          startNode = textNodes[i].parentNode;
+        }
+        startNodeOffset = start - currentCharCount;
+      }
+    }
+    // Adds class/data/events to the nodes between the start and end
+    else if (foundStart && end >= endCharCount) {
+      // Make sure it has its own parent span
+      if (textNodes[i].parentNode.parentNode === node) {
+        var temp_node = document.createElement("span");
+        temp_node.innerHTML = textNodes[i].textContent;
+        textNodes[i].parentNode.replaceChild(temp_node, textNodes[i]);
+        textNodes[i] = temp_node.childNodes[0];
+      }
+      var glow_depth = textNodes[i].parentNode.getAttribute("data-annotationDepth");
+      textNodes[i].parentNode.setAttribute("data-annotationDepth",
+        glow_depth == null ? "1" : (parseInt(glow_depth, 10) + 1).toString());
+      textNodes[i].parentNode.setAttribute("data-annotationID" +
+        annotationId.toString(), annotationId.toString());
+
+      textNodes[i].parentNode.addClass('source_code_glowing_' +
+        (glow_depth == null ? "1" : (parseInt(glow_depth, 10) + 1)));
```

Comment:
Identifier 'glow_depth' is not in camel case.<br>Use '===' to compare with 'null'.